drivers/sgp30/sgp30.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sgp30
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the sensors
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ *
+ * @}
+ */
+#include <string.h>
+
+#include "sgp30.h"
+#include "sgp30_constants.h"
+#include "sgp30_params.h"
+
+#include "checksum/crc8.h"
+#include "ztimer.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+/**
+ * @name      Address pointer values for all SGP30 I2C commands
+ * @{
+ */
+typedef enum {
+    SGP30_CMD_INIT_AIR_QUALITY          = 0x2003,   /**< Start measurement mode */
+    SGP30_CMD_MEASURE_AIR_QUALITY       = 0x2008,   /**< Measure air quality values */
+    SGP30_CMD_GET_BASELINE              = 0x2015,   /**< Get baseline values */
+    SGP30_CMD_SET_BASELINE              = 0x201e,   /**< Set baseline values CMD */
+    SGP30_CMD_SET_HUMIDITY              = 0x2061,   /**< Set absolute humidity value
+                                                         for compensation */
+    SGP30_CMD_MEASURE_TEST              = 0x2032,   /**< Perform on-chip self test */
+    SGP30_CMD_GET_FEATURE_SET_VERSION   = 0x202f,   /**< Get feature set version */
+    SGP30_CMD_MEASURE_RAW_SIGNALS       = 0x2050,   /**< Read raw H2 and Ethanol signals */
+    SGP30_CMD_READ_SERIAL               = 0x3682,   /**< Read serial number */
+    SGP30_CMD_SOFT_RESET                = 0x0006,   /**< Perform General Call reset */
+} sgp30_cmd_t;
+/** @} */
+
+/**
+ * @name      Max delays for SGP30 I2C commands completion
```

Comment:
It looks like this needs mostly milliseconds, can the driver be made to only use ZTIMER_MSEC?

Reply:
hey, let's do this as follow up, it would be an optimization anyways